tests/dip_sip/test_dip_sip.py

```diff
@@ -0,0 +1,168 @@
+    # minigraph facts
+    src_lag = mg_facts['minigraph_portchannel_interfaces'][2]['attachto']
+    dst_lag = mg_facts['minigraph_portchannel_interfaces'][0]['attachto']
+    facts['src_port'] = src_lag
+    facts['dst_port'] = dst_lag
+    logger.info("src_lag is {}, dst_lag is {}".format(src_lag, dst_lag))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][src_lag]['members'][0]]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_lag]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_lag]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_lag]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_lag]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][src_lag]['members'][0]]]
+
+    return facts
+
+
+def port_facts(dut):
+    facts = {}
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
+
+    if not mg_facts['minigraph_interfaces']:
+        pytest.fail("minigraph_interfaces is not defined.")
+    host_facts = dut.setup()['ansible_facts']
+    # minigraph facts
+    src_port = mg_facts['minigraph_interfaces'][2]['attachto']
+    dst_port = mg_facts['minigraph_interfaces'][0]['attachto']
+    facts['src_port'] = src_port
+    facts['dst_port'] = dst_port
+    logger.info("src_port is {}, dst_port is {}".format(src_port, dst_port))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[dst_port]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[src_port]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_port]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_port]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_port]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_port]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][dst_port]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][src_port]]
+
+    return facts
+
+
+@pytest.fixture(scope='function')
+def gather_facts(testbed_devices, testbed):
+    facts = {}
+    topo = testbed['topo']['name']
+    if topo not in TOPO_LIST:
+        pytest.skip("Unsupported topology")
+    logger.info("Gathering facts on DUT ...")
+    dut = testbed_devices["dut"]
+    if topo in PORTS_TOPO:
```

Comment:
https://github.com/Azure/sonic-mgmt/blob/master/tests/conftest.py#L70

Reply:
@lguohan 
get_testbed_type() returns the group of the topology, but in this test t0 and t1-lag are in the same group (lag facts), and t1 is in another group (port facts).
So I believe this function will not be relevant here.

Reply:
i get it. then, we should inspect the topology to understand if it has lag in the topology or not. I think we should remove the hardware coded topology here.

Reply:
@lguohan Done.